### PR TITLE
bugfixes for 1.3.1

### PR DESCRIPTION
**Pull request by e5l**
- revert LocalFileContent
- User-Agent fix
- add coverage
- support logging of `Input` request

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -180,8 +180,11 @@ allprojects {
             }
         }
     }
+
```
Comment (dmitrievanthony) [inline build.gradle]: It doesn't look like a necessary change here.

File: ktor-client/ktor-client-core/common/src/io/ktor/client/engine/Utils.kt [hunk 2]
```diff
@@ -80,3 +81,5 @@ internal suspend inline fun attachToUserJob(callJob: Job) {
         cleanupHandler.dispose()
     }
 }
+
+private fun needUserAgent(): Boolean = !PlatformUtils.IS_BROWSER
```
Comment (dmitrievanthony) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/engine/Utils.kt]: Does it make sense to make it `inline`?

File: ktor-utils/common/src/io/ktor/util/ByteChannels.kt [hunk 3]
```diff
@@ -42,6 +42,33 @@ fun ByteReadChannel.split(coroutineScope: CoroutineScope): Pair<ByteReadChannel,
...
+ */
+@InternalAPI
+fun ByteReadChannel.copyToBoth(first: ByteWriteChannel, second: ByteWriteChannel) {
+    GlobalScope.launch(Dispatchers.Unconfined) {
+        try {
+            while (!isClosedForRead && (!first.isClosedForWrite || !second.isClosedForWrite)) {
+                readRemaining(CHUNK_BUFFER_SIZE).use {
+                    try {
+                        first.writePacket(it.copy())
+                    } catch (_: Throwable) {
```
Comment (dmitrievanthony) [inline ktor-utils/common/src/io/ktor/util/ByteChannels.kt]: Do we intentionally miss all exceptions here?

File: build.gradle [hunk 4]
```diff
@@ -256,3 +259,70 @@ afterEvaluate {
...
+        sourceDirectories.from files(coverageSourceDirs)
+        additionalSourceDirs.from files(coverageSourceDirs)
+        executionData.from files("${buildDir}/jacoco/jvmTest.exec")
+
+        reports {
+            xml.enabled true
+            csv.enabled false
+            html.enabled true
+
+            html.destination file("${buildDir}/jacoco-reports/html")
```
Comment (cy6erGn0m) [inline build.gradle]: Doesn't Gradle have reports directory for such purposes?

File: ktor-utils/common/src/io/ktor/util/PlatformUtils.kt [hunk 5]
```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.util
+
+@InternalAPI
+expect object PlatformUtils {
```
Comment (cy6erGn0m) [inline ktor-utils/common/src/io/ktor/util/PlatformUtils.kt]: This is definitely a thing that everyone would abuse :skull_and_crossbones:

File: ktor-utils/common/src/io/ktor/util/ByteChannels.kt (hunk 3 above)
Comment (e5l) [reply ktor-utils/common/src/io/ktor/util/ByteChannels.kt]: Yep, it's only for logging purposes

File: ktor-utils/common/src/io/ktor/util/PlatformUtils.kt (hunk 5 above)
Comment (e5l) [reply ktor-utils/common/src/io/ktor/util/PlatformUtils.kt]: Probably, but we also can't live without it :)

File: build.gradle (hunk 4 above)
Comment (e5l) [reply build.gradle]: I faced an issue when Gradle overwrite this directory in parrallel

File: build.gradle (hunk 1 above)
Comment (e5l) [reply build.gradle]: tnx

**Pull request merged**